receiver/snmpreceiver/config.go

```diff
@@ -563,13 +591,38 @@ func validateResourceAttributeConfigs(cfg *Config) error {
 		return nil
 	}
 
-	// Make sure each Resource Attribute has either an OID or IndexedValuePrefix
+	// Make sure each Resource Attribute has exactly one of OID or ScalarOID or IndexedValuePrefix, and check that scalar and column OIDs end in the right digit
 	for attrName, attrCfg := range resourceAttributes {
-		if attrCfg.OID == "" && attrCfg.IndexedValuePrefix == "" {
-			combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgResourceAttributeNoOIDOrPrefix, attrName))
+
+		hasOID := attrCfg.OID != ""
+		hasScalarOID := attrCfg.ScalarOID != ""
+		hasIVP := attrCfg.IndexedValuePrefix != ""
+
+		switch {
+		case hasOID:
+			if hasScalarOID || hasIVP {
+				combinedErr = errors.Join(combinedErr, fmt.Errorf(errMsgMultipleKeysSetOnResourceAttribute, attrName))
```

Comment:
Should we continue if this happens?

Reply:
That would make sense, but also skips the second error check which will usually be relevant even after the first one fails. It feels nice to give both at once.